tests/components/modbus/test_modbus_sensor.py

```diff
@@ -540,12 +545,29 @@ async def test_all_sensor(hass, cfg, regs, expected):
         (
             {
                 CONF_COUNT: 1,
+                CONF_REGISTER_SIZE: 2,
                 CONF_PRECISION: 0,
                 CONF_DATA_TYPE: DATA_TYPE_INT,
             },
             [0x0101],
             "257",
         ),
+        (
+            {
+                CONF_COUNT: 2,
+                CONF_REGISTER_SIZE: 4,
+                CONF_PRECISION: 0,
+                CONF_DATA_TYPE: DATA_TYPE_CUSTOM,
+                CONF_STRUCTURE: ">2i",
+            },
+            [0x0000, 0x0100, 0x0000, 0x0032],
+            "256,50",
+        ),
+        (
+            {CONF_COUNT: 1, CONF_REGISTER_SIZE: 2},
+            [0x0101],
+            "257",
+        ),
     ],
 )
 async def test_struct_sensor(hass, cfg, regs, expected):
```

Comment:
You missing to test that Swap works with register_size 4. I think it needs to be part extension of the existing test and part a new one.